### ISSUE TITLE
`B008` doesn't report when the annotation refers to an immutable type, but `RUF009` does

### ISSUE THREAD

**Issue by InSyncWithFoo** [astral-sh/ruff#15772]
Minimal reproducible example ([playground](https://play.ruff.rs/d559f6aa-7617-4e99-97a0-57198b791c5f)):

```python
def _(
    this_is_fine: int = f(),           # No error
    this_is_not: list[int] = f()       # B008: Do not perform function call `f` in argument defaults
): ...


@dataclass
class _:
    this_is_not_fine: list[int] = f()  # RUF009: Do not perform function call `f` in dataclass defaults
    this_is_also_not: int = f()        # RUF009: Do not perform function call `f` in dataclass defaults
```

[`RUF009`](https://docs.astral.sh/ruff/rules/function-call-in-dataclass-default-argument/) should behave similar to [`B008`](https://docs.astral.sh/ruff/rules/function-call-in-default-argument/) and ignore attributes with immutable types.

**Comment by dhruvmanila** [comment]
Yeah, this seems reasonable. I think it's because we don't look at the annotation which can be done using `is_immutable_annotation`.

**Comment by smokyabdulrahman** [comment]
I hope [this PR](https://github.com/astral-sh/ruff/pull/16048) resolved the issue.
